cpp/src/gandiva/llvm_generator.cc

```diff
@@ -588,15 +561,15 @@ void LLVMGenerator::Visitor::Visit(const LiteralDex& dex) {
 void LLVMGenerator::Visitor::Visit(const NonNullableFuncDex& dex) {
   ADD_VISITOR_TRACE("visit NonNullableFunc base function " +
                     dex.func_descriptor()->name());
-  LLVMTypes* types = generator_->types_.get();
+  LLVMTypes& types = generator_->types();
```

Comment:
Better to avoid mutable references for consistency with rest of codebase. see also https://google.github.io/styleguide/cppguide.html#Reference_Arguments